krystex/src/main/java/com/flipkart/krystal/krystex/node/Node.java

```diff
@@ -602,11 +678,17 @@ private NavigableSet<MainLogicDecorator> getSortedDecorators(DependantChain depe
   private record DependencyNodeExecutions(
       LongAdder executionCounter,
       Set<ResolverDefinition> executedResolvers,
+      Map<RequestId, Boolean> skippedDependency,
```

Comment:
Does this need to be a map? One object of this type represents all the dependencyNodeExecutions for a given incoming requestId. My understanding is that either all the dependencyExecutions will be skipped, or none. If this is correct, then this just needs to be a boolean